### PR TITLE
Misc improvements

### PR DESCRIPTION
**Pull request by eldritchconundrum**

### CODE REVIEW

File: src/analyzer.fs [hunk 1]
```diff
@@ -74,10 +74,10 @@ module private VariableInlining =
             if not ident.DoNotInline && not ident.ToBeInlined && not ident.VarDecl.Value.isEverWrittenAfterDecl then
                 match localReferences.TryGetValue(def.Key), allReferences.TryGetValue(def.Key) with
                 | (_, 1), (_, 1) when isConst ->
-                    debug $"inlining local variable '{Printer.debugIdent ident}' because it's safe to inline (const) and used only once"
+                    debug $"{ident.Loc}: inlining local variable '{Printer.debugIdent ident}' because it's safe to inline (const) and used only once"
```
Comment (laurentlb) [inline src/analyzer.fs:77]: Should this be a new argument for debug?

**Review by laurentlb — APPROVED** [review]

File: src/analyzer.fs (hunk 1 above)
Comment (eldritchconundrum) [reply src/analyzer.fs:77]: What would be the use of debug information without line numbers?

Comment (laurentlb) [reply src/analyzer.fs:77]: Debug could have two arguments: loc + msg, so that they are both mandatory.

Comment (eldritchconundrum) [reply src/analyzer.fs:77]: We have some messages that don't have a loc.
But, debug could accept a Loc option, even just as a reminder to use a loc when we have one.

**Pull request merged**